ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -1145,6 +1110,142 @@ public void shutdown() throws StoreException {
     shutdown(false);
   }
 
+  /**
+   * Update the sealed status of the replica.
+   */
+  private void updateSealedStatus() {
+    // In two zk clusters case, "replicaId.isSealed()" might be true in first cluster but is still unsealed in second
+    // cluster, as server only spectates the first cluster. To guarantee both clusters have consistent seal/unseal
+    // state, we bypass "isSealed()" check if there are more than one replicaStatusDelegates.
+    ReplicaSealStatus replicaSealStatus1 = replicaSealStatus.get();
+    if (resolveReplicaSealStatusFromLogSize() == ReplicaSealStatus.SEALED && (
+        replicaSealStatus.get() != ReplicaSealStatus.SEALED || (replicaStatusDelegates.size() > 1
```

Comment:
Should this be `replicaId.isSealed()` since `replicaSealStatus.get()` is local value?

Reply:
Yeah, I think this should be `replicaId.isSealed()` since `replicaSealStatus` is only updated during two ZK clusters case